client/client_test.go

```diff
@@ -83,6 +83,30 @@ func TestPing(t *testing.T) {
 			t.Error("Round-trip time returned on failure should've been 0")
 		}
 	}
+	if success, rtt := Ping("::1", &Config{Timeout: 500 * time.Millisecond, Network: "ip4"}); success {
+		t.Error("expected false, because the IP isn't an IPv4 address")
+		if rtt != 0 {
+			t.Error("Round-trip time returned on failure should've been 0")
+		}
+	}
+	if success, rtt := Ping("127.0.0.1", &Config{Timeout: 500 * time.Millisecond, Network: "ip6"}); success {
+		t.Error("expected false, because the IP isn't an IPv6 address")
+		if rtt != 0 {
+			t.Error("Round-trip time returned on failure should've been 0")
+		}
+	}
+	if success, rtt := Ping("ipv6.google.com", &Config{Timeout: 500 * time.Millisecond, Network: "ip6"}); !success {
+		t.Error("expected true, because the host is IPv6-only")
+		if rtt == 0 {
+			t.Error("Round-trip time returned on success should've higher than 0")
+		}
+	}
+	if success, rtt := Ping("ipv4.google.com", &Config{Timeout: 500 * time.Millisecond, Network: "ip4"}); !success {
+		t.Error("expected true, because the host is IPv4-only")
```

Comment:
```
--- FAIL: TestPing (1.34s)
    client_test.go:99: expected true, because the host is IPv6-only
    client_test.go:101: Round-trip time returned on success should've higher than 0
    client_test.go:105: expected true, because the host is IPv4-only
```
Looks like the integration tests are failing 😔

Reply:
I added those two unit tests, because they're for IPv6-only and IPv4-only domains, respectively and it would have been an ideal check to confirm functionality.
Due to the apparent limitation of ICMP being dropped in the network of GitHub-hosted runners (just confirmed this in a test repo) we have two options:

1. Drop these two tests
2. Use a domain that is single-stacked and points to only `::1` or `127.0.0.1` using e.g. sslip.io or backname.io (adds maybe unwanted dependence on theses services)

What would you prefer?

Reply:
Ah, this is tricky.. At least the local icmp check works.

I'm okay with commenting the two integration tests as well as adding a short comment explaining why we can't perform integration tests, however, could you add an additional test verifying the default behavior (neither ip6 nor ip4 -- so the default `ip`) works?

At least this should ensure that future changes do not alter the default behavior.

Reply:
Sounds good, like this?

> Also, just want to point out that I appreciate you taking the time to be so thorough :)

🤝